# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/chrome-webidl/WindowGlobalActors.webidl
# Commit: 6a0dcd8f4e3e
# Full Hash: 6a0dcd8f4e3e35d27edba6a4fef795cf8174ac78
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2021-02-18 04:16:06
# Regressor Bug: 1692350
# File Overlap Count: 1
# Description:
#   Bug 1692350 - Implement a telemetry probe to collect page load data for documents that use lazyload r=emilio,chutten
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104912
# ==============================================================================

diff -r 6d7590bfd8d3 -r 6a0dcd8f4e3e dom/chrome-webidl/WindowGlobalActors.webidl
--- a/dom/chrome-webidl/WindowGlobalActors.webidl	Wed Feb 17 17:08:51 2021 +0000
+++ b/dom/chrome-webidl/WindowGlobalActors.webidl	Wed Feb 17 17:17:52 2021 +0000
@@ -25,6 +25,10 @@
 
   // True if the principal of this window is for a local ip address.
   readonly attribute boolean isLocalIP;
+
+  // True if the corresponding document has `loading='lazy'` images;
+  // It won't become false if the image becomes non-lazy.
+  readonly attribute boolean hadLazyLoadImage;
 };
 
 // Keep this in sync with nsIContentViewer::PermitUnloadAction.